script.embuary.helper/resources/lib/utils.py

```diff
@@ -0,0 +1,392 @@
+                        properties=tvshow_properties,
+                        params={'tvshowid': int(tvshow_id)}
+                        )
+
+    try:
+        details = tvshow_query['result']['tvshowdetails']
+    except Exception:
+        log('Show details by season: Could not get TV show details')
+        return
+
+    if int(details['episode']) > int(details['watchedepisodes']):
+        unwatchedepisodes = int(details['episode']) - int(details['watchedepisodes'])
+        unwatchedepisodes = str(unwatchedepisodes)
+    else:
+        unwatchedepisodes = '0'
+
+    winprop('tvshow.dbid', str(details['tvshowid']))
+    winprop('tvshow.rating', str(round(details['rating'],1)))
+    winprop('tvshow.seasons', str(details['season']))
+    winprop('tvshow.episodes', str(details['episode']))
+    winprop('tvshow.watchedepisodes', str(details['watchedepisodes']))
+    winprop('tvshow.unwatchedepisodes', unwatchedepisodes)
+
+
+def hyperion_winscreencap(params):
+
+    port = params.get('port', '9192')
+    action = params.get('cmd').upper()
+
+    try:
+        response = urllib.urlopen('http://localhost:%s?command=%s&force=True' % (port,action))
+        response.read()
+        response.close()
+    except:
+        pass
+
+
+class PlayCinema(object):
+
+    def __init__(self, params):
+
+        self.trailer_count = xbmc.getInfoLabel('Skin.String(TrailerCount)') if not xbmc.getInfoLabel('Skin.String(TrailerCount)') == '0' else ''
+        self.intro_path = xbmc.getInfoLabel('Skin.String(IntroPath)')
+
+        self.dbid = params.get('dbid')
+        self.dbtype = params.get('type')
+        self.item_title = remove_quotes(params.get('title'))
+
+        if not self.dbid or not self.item_title or not self.dbtype:
+            for i in range(30):
+
+                if xbmc.getInfoLabel('Container.ListItem.Label'):
+                    break
+                xbmc.sleep(100)
+
+            self.dbid = xbmc.getInfoLabel('Container.ListItem.DBID')
+            self.item_title = xbmc.getInfoLabel('Container.ListItem.Label')
+            self.dbtype = xbmc.getInfoLabel('Container.ListItem.DBTYPE')
+
+        if self.dbid and self.item_title and self.dbtype:
+            log('Play with cinema mode: %s' % self.item_title)
+            self.run()
+        else:
+            log('Play with cinema mode: Not enough arguments')
+
+
+    def run(self):
+        index = 0
+        VIDEOPLAYLIST.clear()
+
+        if self.trailer_count:
+            trailers = self.get_trailers()
+            for trailer in trailers:
+
+                trailer_title = '%s (%s)' % (trailer['title'], xbmc.getLocalizedString(20410))
+                trailer_rating = str(round(trailer['rating'],1))
+                thumbnailImage = trailer['art'].get('landscape') or trailer['art'].get('fanart') or trailer['art'].get('poster', '')
+
+                listitem = xbmcgui.ListItem(trailer_title, thumbnailImage=thumbnailImage)
```

Comment:
thumbnailImage is deprecated please use setArt only

Reply:
Will be done